packages/flutter/lib/src/widgets/localizations.dart

```diff
@@ -494,9 +504,8 @@ class _LocalizationsState extends State<Localizations> {
         setState(() {
           _typeToResources = value;
           _locale = locale;
+          _loadGeneration += 1;
```

Comment:
you're no longer actually calling dispatchDidChangeDependencies in the synchronous future case.